Makefile

```
@@ -1,16 +1,28 @@
-
 CC=gcc
-CFLAGS=-O3 -march=native -std=c11 -Wall -Wno-stringop-truncation
-lc: lc.c
-	$(CC) -o lc $(CFLAGS) $<
+CFLAGS=-Og -g -march=native -std=c11 -Wall -Wno-stringop-truncation -I.
+LIBS = lclib.h
+OBJS = lc.o lclib.o
+MAIN = lc.c
+
+%.o: %.c $(LIBS) #libs = objects confirmed
+	$(CC) -c -o $@ $(CFLAGS) $< 
+
+lc: $(OBJS)
+	$(CC) -o lc -I $(CFLAGS) $^
 
 debug-lc: lc.c
 	$(CC) -o lc-debug -Og -g $<
 
 check:
 	echo "check"
+	valgrind --leak-check=full ./lc --help
 
+suid: lc
+	# I wouldn't use this willy-nilly
```

What do you think of the below, i think it would be valuable to expand on the issue
for the off-chance that someone ever use this. And also to echo it, so we are sure users 
could at least have seen it. 
```suggestion
	echo "I wouldn't use this willy-nilly, as"
	echo "there is a chance for an uncaught"
	echo "buffer overflow attack vector that"
	echo "could lead to privilege escalation"
	echo "i.e. attack can do admin stuff"
```